packages/ubuntu_desktop_installer/test/connect_to_internet/hidden_wifi_model_test.dart

```diff
@@ -0,0 +1,161 @@
+import 'dart:async';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/connect_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/hidden_wifi_model.dart';
+import 'package:ubuntu_desktop_installer/pages/connect_to_internet/wifi_model.dart';
+import 'package:ubuntu_desktop_installer/services/network_service.dart';
+
+import 'hidden_wifi_model_test.mocks.dart';
+
+@GenerateMocks([
+  NetworkManagerAccessPoint,
+  NetworkManagerActiveConnection,
+  NetworkManagerDevice,
+  NetworkManagerDeviceWireless,
+  NetworkManagerSettingsConnection,
+  NetworkService,
+])
+void main() {
+  late HiddenWifiModel model;
+  late WifiDevice wifi;
+  late NetworkService service;
+  late NetworkManagerDevice device;
+  late NetworkManagerDeviceWireless wireless;
+  late NetworkManagerAccessPoint ap;
+  late StreamController<List<String>> serviceChanged;
+  late StreamController<List<String>> deviceChanged;
+  late StreamController<List<String>> wirelessChanged;
+  late StreamController<List<String>> accessPointChanged;
+
+  setUp(() async {
+    service = MockNetworkService();
+    serviceChanged = StreamController<List<String>>(sync: true);
+    when(service.propertiesChanged).thenAnswer((_) => serviceChanged.stream);
+    when(service.wirelessDevices).thenReturn([]);
+    when(service.wirelessEnabled).thenReturn(true);
+
+    device = MockNetworkManagerDevice();
+    when(device.udi).thenReturn('test udi');
+    when(device.hwAddress).thenReturn('test address');
+    deviceChanged = StreamController<List<String>>.broadcast(sync: true);
+    when(device.propertiesChanged).thenAnswer((_) => deviceChanged.stream);
+    when(device.state).thenReturn(NetworkManagerDeviceState.activated);
+
+    ap = MockNetworkManagerAccessPoint();
+    accessPointChanged = StreamController<List<String>>.broadcast(sync: true);
+    when(ap.propertiesChanged).thenAnswer((_) => accessPointChanged.stream);
+    when(ap.ssid).thenReturn('foo'.codeUnits);
+    when(ap.hwAddress).thenReturn('test address');
+
+    wireless = MockNetworkManagerDeviceWireless();
+    when(device.wireless).thenReturn(wireless);
+    wirelessChanged = StreamController<List<String>>.broadcast(sync: true);
+    when(wireless.propertiesChanged).thenAnswer((_) => wirelessChanged.stream);
+    when(wireless.accessPoints).thenReturn([]);
+    when(wireless.activeAccessPoint).thenReturn(null);
+
+    model = HiddenWifiModel(service);
+    await model.init();
+    wifi = WifiDevice(device);
+  });
```

Comment:
pretty poor testability with this amount of setup boilerplate needed, but it would be a separate task to see what we could do about it...